File: TowerOfHanoi.swift
var count: Int = 0
func TOH(nod: Int, sour: String, aux: String, dest: String) {
  if nod == 0 {
    return
  }
  TOH(nod: nod - 1, sour: sour, aux: dest, dest: aux)
  print("Move disk \(nod) from tower \(sour) to tower \(dest)")
  count += 1
  TOH(nod: nod - 1, sour: aux, aux: sour, dest: dest)
}


TOH(nod: 6, sour: "A", aux: "B", dest: "C")
print("Total steps: \(count)")

File: TowerOfHanoi.java
public class Solution {
    ArrayList<ArrayList<Integer>> operation = new ArrayList<ArrayList<Integer>>();
    public ArrayList<ArrayList<Integer>> towerOfHanoi(int A) {
        TOH(A, 1, 2, 3);
        return operation;
    }

    void TOH(int a, int sour, int aux, int dest) {
        if (a < 0) {
            return;
        }
        if (a == 0) {
            return;
        }
        TOH(a-1, sour, dest, aux);
        ArrayList<Integer> op = new ArrayList<>();
        op.add(a);
        op.add(sour);
        op.add(dest);
        operation.add(op);
        TOH(a-1, aux, sour, dest);
    }
}
